metaci/release/tests/test_models.py

```diff
@@ -30,3 +31,67 @@ def test_empty_release_init(self):
         assert not release.work_item_link  # checking default set to None
         assert release.change_case_template  # checking default see not None
         assert not release.change_case_link  # checking default set to None
+
+    ####### Keep getting the following error: ###############
+    # ValueError: save() prohibited to prevent data loss due to unsaved related object 'release'.
+    #########################################################
+    # def test_release_implementation_steps_plan_role(self):
+    #     release = Release(
+    #         repo=RepositoryFactory(
+    #             default_implementation_steps={
+    #                 "implementation_steps": [
+    #                     {
+    #                         "plan": "release",
+    #                         "duration": 10,
+    #                         "start_time": 8,
+    #                         "start_date_offset": 0,
+    #                     },
+    #                 ]
+    #             },
+    #         ),
+    #         change_case_template=ChangeCaseTemplate(),
+    #     )
+    #     release.implementation_steps.set(
+    #         [
+    #             ImplementationStep(
+    #                 release=ReleaseFactory(
+    #                     repo=RepositoryFactory(
+    #                         default_implementation_steps={
+    #                             "implementation_steps": [
+    #                                 {
+    #                                     "plan": "release",
+    #                                     "duration": 10,
+    #                                     "start_time": 8,
+    #                                     "start_date_offset": 0,
+    #                                 },
+    #                             ]
+    #                         },
+    #                     )
+    #                 ),
+    #                 plan=PlanFactory(),
+    #                 external_id="1000",
+    #                 start_time="2021-06-08T08:00:00+00:00",
+    #                 stop_time="2021-06-09T18:00:00+00:00",
+    #             ),
+    #         ],
+    #         bulk=False,
+    #     )
+    #     assert release.create_default_implementation_step("test") == None
```

Comment:
how can I test this without receiving  `ValueError: save() prohibited to prevent data loss due to unsaved related object 'release'.` I've attempted to use release.implementation_steps.set() with no success, any ideas @davisagli ?

Reply:
@jofsky can I see the stacktrace so I can see exactly where that error is happening?